lib/buildkite/pipelines/command.rb

```diff
@@ -17,15 +19,25 @@ def pipeline(subcommand, *args)
         end
 
         def artifact(subcommand, *args)
+          case subcommand.to_s
+          when 'shasum', 'search'
+            new(:'artifact', subcommand, *args).run(capture: true)
+          else
           new(:artifact, subcommand, *args).run
+          end
         end
 
         def annotate(body, *args)
           new(:annotate, body, *args).run
         end
 
         def meta_data(subcommand, *args)
-          new(:'meta-data', subcommand, *args).run
+          case subcommand.to_s
+          when 'get', 'keys'
+            new(:'meta-data', subcommand, *args).run(capture: true)
+          else
+            new(:'meta-data', subcommand, *args).run
+          end
```

Comment:
It looks like you’re casing on whether or not to capture, so maybe the case statement should just return true or false? Then you can call the run method once and pass that Boolean through.